src/hasSubObject.test.ts

```diff
@@ -96,15 +96,35 @@ describe("typing", () => {
       const obj = {} as { a?: string; b?: number };
 
       if (hasSubObject(obj, {})) {
```

Comment:
@eranhirsch In #683, I used a block statement ([link](https://github.com/remeda/remeda/pull/686/files#diff-aa20fb5170cbafcaeb26ac51cb81c5c91a4de4f0b89fa3d7d42fb7f34546a091R158-R192)) to isolated each case since there are a lot of cases and it's hard to name. Will this work for this PR as well?

Reply:
I haven't looked at the other PR yet, I prefer each PR be a standalone thing in terms of code quality, as we might not ship that other PR at the same time. I prefer splitting the tests along the lines I mentioned, I like each test have a few asserts, the whole point of tests is to split asserts on logical boundaries.

Reply:
Done. Also changed all `test` to `it` to make it more similar to how `hasAtLeast.test.ts` is written.